test/test_rng.cpp

```diff
@@ -436,4 +437,46 @@ TEST_F(RNGTest, FunctionalUniform) {
   }
 }
 
+namespace {
+
+int64_t get_current_offset() {
+  auto gen = at::cuda::detail::getDefaultCUDAGenerator();
+  {
+    // See Note [Acquire lock when using random generators]
+    std::lock_guard<std::mutex> lock(gen.mutex());
+    auto philox_args =
+        at::check_generator<at::CUDAGeneratorImpl>(gen)->philox_cuda_state(0);
+    auto seeds = at::cuda::philox::unpack(philox_args);
+    return std::get<1>(seeds);
+  }
+}
+
+} // namespace
+
+TEST_F(RNGTest, DifferentOffsets) {
+  // Check that multiple runs of RNG kernel does not produce the same numbers,
+  // and it does bump up RNG offset for ATen.
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  Val* size_val = IrBuilder::create<Val>(DataType::Int);
+  fusion->addInput(size_val);
+  TensorView* tv0 = rand({size_val}, DataType::Float);
+  fusion->addOutput(tv0);
+
+  FusionExecutorCache fec(std::move(fusion_ptr));
+
+  for (int64_t size : {1, 4}) {
+    at::manual_seed(0);
+    EXPECT_TRUE(get_current_offset() == 0);
+    auto r1 = fec.runFusionWithInputs({size}).at(0);
+    EXPECT_TRUE(get_current_offset() == 4);
+    auto r2 = fec.runFusionWithInputs({size}).at(0);
+    EXPECT_TRUE(get_current_offset() == 8);
```

Comment:
Should we also test with other dtypes and check that offset moves by the right size amount?

Reply:
Thank you for the suggestion. I don't think the number of offsets moved depend on dtype, because the dtype difference is realized on the `subsequence` space, not on the `offset` space. However, offsets do depend on the number of random ops in the fusion. So I added another fusion with two random ops, and checked that the "current offset" is correctly moved by 2*4 instead of just 4.